### PR TITLE
Added empty string check

### PR DESCRIPTION
**Pull request by SemperFidelisTyrannosaurus**

### CODE REVIEW

**Comment by drivenflywheel** [comment]
Consider permanently adding a multi-step entry to the bottom of the `magic` file, along with a corresponding UNKNOWN@2.dat file in src/test/resources/emissary/id/UnixFilePlaceTest that triggers the partial match.  If you want to overachieve, add a unit test that leverages the `LogbackTester` class to demonstrate that the exception is no longer thrown/logged.

**Comment by SemperFidelisTyrannosaurus** [comment]
Added additional unit test to make sure exception wasn't thrown.  Unfortunately it doesn't seem like there is a way to confirm that there are no exceptions with the undepricated method in LogbackTester.  That should be fixed but may be outside the scope of this ticket.

**Comment by drivenflywheel** [comment]
> ... it doesn't seem like there is a way to confirm that there are no exceptions with the undepricated method in LogbackTester. 

`logbackTester.checkLogList(Collections.emptyList());` ?

**Comment by SemperFidelisTyrannosaurus** [comment]
> > ... it doesn't seem like there is a way to confirm that there are no exceptions with the undepricated method in LogbackTester.
> 
> `logbackTester.checkLogList(Collections.emptyList());` ?

Doh, misread the Validation at the the beginning of the method and didn't think to test it for myself.  updated.

File: src/test/java/emissary/id/UnixFilePlaceTest.java [hunk 1]
```diff
@@ -1,21 +1,54 @@
...
+        try (LogbackTester logbackTester = new LogbackTester(UnixFilePlace.class.getName())) {
+            try (InputStream doc = new ResourceReader().getResourceAsStream(resource)) {
+                byte[] data = IOUtils.toByteArray(doc);
+                IBaseDataObject payload = DataObjectFactory.getInstance(data, resource, Form.UNKNOWN);
+                processPreHook(payload, resource);
+                place.agentProcessHeavyDuty(payload);
+                processPostHook(payload, resource);
+            } catch (Exception ex) {
+                logger.error("Error running test {}", resource, ex);
+                fail("Cannot run test " + resource, ex);
```
Comment (drivenflywheel) [inline src/test/java/emissary/id/UnixFilePlaceTest.java:47]: Why "Cannot run test"?  This code can only be reached by running the test

File: src/test/java/emissary/id/UnixFilePlaceTest.java [hunk 2]
```diff
@@ -1,21 +1,54 @@
...
     }
 
+
     @Override
     public IServiceProviderPlace createPlace() throws IOException {
         return new UnixFilePlace();
     }
+
+    @Test
+    void testException() throws IOException {
```
Comment (drivenflywheel) [inline src/test/java/emissary/id/UnixFilePlaceTest.java:35]:
```suggestion
    void testMultiStepRuleAtBottomOfMagicFile() throws IOException {
```

File: src/test/java/emissary/id/UnixFilePlaceTest.java [hunk 3]
```diff
@@ -1,21 +1,54 @@
...
+
+        String resource = "emissary/id/UnixFilePlaceTest/UNKNOWN@2.dat";
+        try (LogbackTester logbackTester = new LogbackTester(UnixFilePlace.class.getName())) {
+            try (InputStream doc = new ResourceReader().getResourceAsStream(resource)) {
+                byte[] data = IOUtils.toByteArray(doc);
+                IBaseDataObject payload = DataObjectFactory.getInstance(data, resource, Form.UNKNOWN);
+                processPreHook(payload, resource);
```
Comment (drivenflywheel) [inline src/test/java/emissary/id/UnixFilePlaceTest.java:42]: Remove unnecessary call
```suggestion
```

File: src/test/java/emissary/id/UnixFilePlaceTest.java [hunk 4]
```diff
@@ -1,21 +1,54 @@
...
+                place.agentProcessHeavyDuty(payload);
+                processPostHook(payload, resource);
```
Comment (drivenflywheel) [inline src/test/java/emissary/id/UnixFilePlaceTest.java:44]: Remove unnecessary call
```suggestion
```

Comment (drivenflywheel) [inline src/test/java/emissary/id/UnixFilePlaceTest.java:44]:
```suggestion
                processPostHook(payload, resource);
                assertTrue(StringUtils.isBlank(payload.getProcessingError()), "Expected no processing error");
```

**Review by drivenflywheel — CHANGES_REQUESTED** [review]
Minor cleanup suggested

**Review by drivenflywheel — APPROVED** [review]
🚀

**Review by jpdahlke — APPROVED** [review]

**Review by drivenflywheel — APPROVED** [review]

**Pull request merged**